File: 프로그래머스/lv2/42578. 의상/의상.js
function solution(clothes) {
    var answer = 1;
    const cloth = clothes.reduce((arr,val)=>{
        arr[val[1]] = (arr[val[1]] || 1) + 1;
        return arr;
    },{})
    
    for(let i in cloth){
        answer *= cloth[i];
    }
    return answer - 1;
}

File: 프로그래머스/2/42578. 의상/의상.py
from collections import defaultdict

def solution(clothes):
    answer = 1
    maps = defaultdict(int)
    for cloth, kind in clothes:
        maps[kind] += 1
    
    val = list(maps.values())

    for i in val:
        answer *= (i+1)
    
    return answer - 1